src/Simplex/FileTransfer/Transport.hs

```diff
@@ -214,6 +214,8 @@ data XFTPErrorType
     TIMEOUT
   | -- | bad redirect data
     REDIRECT {redirectError :: String}
+  | -- | cannot proceed with download from unknown server without proxy
+    UNKNOWN_NO_PROXY
```

Comment:
this is the error type that is sent in protocol, it shouldn't be here

Reply:
should be in AgentErrorType, also the name is wrong and too long.

Reply:
AgentErrorType should have FILE constructor and all errors that are not sent via the protocol should be moved here. XFTP constructor in AgentErrorType should only be used for errors sent by servers to clients (and currently these errors are forwarded as INTERNAL, so it also needs to be fixed).